devito/propagator.py

```diff
@@ -362,15 +438,24 @@ def time_substitutions(self, sympy_expr):
         return sympy_expr
 
     def add_time_loop_stencil(self, stencil, before=False):
+        """Add the given stencil to the time loop stencil
```

Comment:
Add a statement either before or after the main spatial loop, but still inside the time loop.  Flag indicating whether the statement should be inserted before.